test/distributed/test_data_parallel.py

```diff
@@ -618,8 +618,7 @@ def test_replicate(self):
         for devices in [(0, 1), [0, 1]]:
             replicas = dp.replicate(module, devices)
             for i, replica in enumerate(replicas):
-                for p in replica.parameters():
-                    self.assertEqual(p.get_device(), i)
+                self.assertRaises(RuntimeError, lambda: print(replica.parameters()))
```

Comment:
shall we check the error message as well?

Reply:
Will reflect this in the next update. Let me know if this isn't what you were thinking:

```
                with self.assertRaises(RuntimeError) as error:
                    print(replica.parameters())
                self.assertEqual(error.exception.message, \
                    "parameters() cannot be called on a replicated model")
```